synapse/handlers/room.py

```diff
@@ -753,6 +761,7 @@ async def create_room(
             invite_list=invite_list,
             initial_state=initial_state,
             creation_content=creation_content,
+            creator=creator,
```

Comment:
I think you need to change the check on `power_level_content_override` above to check the power levels include the *creator* rather than the requester.

Generally I would recommend renaming `user_id` to `requester_user_id` to reduce confusion.